From e24f68b2280ee0c1c86a9a1084ad61e98fc5ee0e Mon Sep 17 00:00:00 2001
From: Xiao,Feng <fengx.xiao@intel.com>
Date: Thu, 4 Jul 2013 18:59:39 +0800
Subject: Modify the rtp-info assert condition in MyHandler to avoid crash

BZ: 116517

After multiple seeking, the play response received from the
rtsp server may not contain some field like "range". It can
casue the assert failed.

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I1ad6b50368064fd688b65a4a1a2f347947f14d94
Orig-Change-Id: I16f27964f6d60df6658d544bc9f818a822e506b4
Signed-off-by: Xiao,Feng <fengx.xiao@intel.com>
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index dfe1bee..210de04 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1271,9 +1271,12 @@ struct MyHandler : public AHandler {
                         timeout->post(kStartupTimeoutUs);
 
                         ssize_t i = response->mHeaders.indexOfKey("rtp-info");
-                        CHECK_GE(i, 0);
+                        if (i < 0) {
+                            ALOGE("rtp-info is %d", i);
+                            result = UNKNOWN_ERROR;
+                        }
 
-                        ALOGV("rtp-info: %s", response->mHeaders.valueAt(i).c_str());
+                        ALOGV("rtp-info: %s", (i >= 0) ? response->mHeaders.valueAt(i).c_str() : "-1");
 
                         ALOGI("seek completed.");
                     }
-- 
1.7.4.1

